src/dune_pkg/opam.ml

```diff
@@ -0,0 +1,188 @@
+open Stdune
+module Package_name = Dune_lang.Package_name
+
+module Repo = struct
+  let ( / ) = Filename.concat
+
+  type t = { packages_dir_path : string }
+
+  let of_opam_repo_dir_path opam_repo_dir_path =
+    if not (Sys.file_exists opam_repo_dir_path) then
+      User_error.raise [ Pp.textf "\"%s\" does not exist" opam_repo_dir_path ];
+    if not (Sys.is_directory opam_repo_dir_path) then
+      User_error.raise
+        [ Pp.textf "\"%s\" is not a directory" opam_repo_dir_path ];
+    let packages_dir_path = opam_repo_dir_path / "packages" in
+    if
+      not
+        (Sys.file_exists packages_dir_path && Sys.is_directory packages_dir_path)
+    then
+      User_error.raise
+        [ Pp.textf
+            "\"%s\" doesn't look like a path to an opam repository as it lacks \
```

Comment:
Should be `%s` and `Path.to_maybe_string_quoted opam_repo_dir_path` to be consistent with our other messages.

Reply:
(using `String.maybe_quoted` which works for `Filename.t`s)